src/merkle.rs

```diff
@@ -0,0 +1,168 @@
+use crate::Example;
+use log::debug;
+use assembly::Assembler;
+use vm_core::{Felt, FieldElement, hasher, Program, ProgramInputs, StarkField};
+use rand_utils::prng_vector;
+
+// EXAMPLE BUILDER
+// ================================================================================================
+
+pub fn get_example(depth: usize) -> Example {
```

Comment:
As is, I think this example is not super interesting. I wonder if we could transform it into a version of [Semaphore protocol](https://github.com/bobbinth/semaphore/blob/main/src/lib.rs). It could take a little bit of effort - but I think it would be very interesting how Semaphore implemented in Miden VM performs against native STARK implementation.

Reply:
@bobbinth The linked repo doesn't seem to be accessible. Is it private?